### DISCUSSION TITLE
Changing background colors of tags

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#1678] [Q&A] [answered]
Where I can change the background color of the tags? And how to use colons ?
Inserting "Regionale Gliederung: Landkreise und kreifreie Städte"
produces this error: 
`  Liquid Exception: undefined method `gsub' for {"Regionale Gliederung"=>"Landkreise und kreisfreie Städte"}:Hash in /_layouts/goal.html
jekyll 3.8.4 | Error:  undefined method `gsub' for {"Regionale Gliederung"=>"Landkreise und kreisfreie Städte"}:Hash
Error: Process completed with exit code 1.`

I have tried "Regionale Gliederung":" Landkreise und kreifreie Städte"
This works but the " " are shown too

![image](https://user-images.githubusercontent.com/94443931/173817858-0b7fafa5-283b-4386-9197-2a03ebf9fa0c.png)

example: https://christoph-lsn.github.io/MT_Site/1/
site repo:  https://github.com/christoph-LSN/MT_Site
data repo; https://github.com/christoph-LSN/IM_Data

**Comment by brockfanning** [comment, marked as answer]
@christoph-LSN We have a "sass variable" for this color, which you can find here: https://github.com/open-sdg/open-sdg/blob/2.1.0-dev/_sass/variables/_colors.scss#L79

So the cleanest way to change this color would be to add a line to your `_sass/variables.scss` file, like:

```
$tag-backgroundColor: pink;
```

Here's a bit more [detail on Sass variables in Open SDG](https://open-sdg.readthedocs.io/en/latest/customisation/#overriding-color-via-sass-variables).

**Reply by christoph-LSN** [reply]
@brockfanning 
Thank you, it works, but now all tags are green. I'd like to have the first tag green, the second blue an the third pink....

**Reply by brockfanning** [reply]
@christoph-LSN If you check the markup, you'll see that each tag gets a custom class according to the content of the name. For example your tag "Datenstand 2020" has a class of "tag-datenstand-2020":

![tag-class](https://user-images.githubusercontent.com/1319083/174607692-e8aa8437-0743-40d7-b2d1-4f14a21ba847.png)


So one possibility would be to pick a particular color for each possible tag:

```
.tag-datenstand-2020 {
  background-color: pink;
}
```

**Reply by christoph-LSN** [reply]
@brockfanning ok, where i have to add these lines...

adding it in my [variable.scss ](https://github.com/christoph-LSN/MT_Site/blob/develop/_sass/variables.scss) produces the following error:

Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/style.scss':
                    Invalid CSS after "...atenstand-2020 ": expected ":", was "{" on line 20
jekyll 3.8.4 | Error:  Invalid CSS after "...atenstand-2020 ": expected ":", was "{" on line 20
Error: Process completed with exit code 1

**Reply by brockfanning** [reply]
@christoph-LSN It would be more correct to put it here: https://github.com/christoph-LSN/MT_Site/blob/develop/_sass/custom.scss

The error, though, is because it should be `.tag-datenstand-2020` instead of `$tag-datenstand-2020`.

**Reply by christoph-LSN** [reply]
@brockfanning I put it in my custom.scss but there is no result. Do I have to leave this line:
`$tag-backgroundColor: yellow;` 
in my variables.scss or do I have to delete it?

**Reply by brockfanning** [reply]
@christoph-LSN It looks like a problem with CSS specificity, which is always a headache with CSS. But it should work if you wrap it in a `.tags` block, like this:

```
.tags {
  .tag-datenstand-2020 {
    background-color: pink;
  }
}
```

Also as you add more, you would put them inside the same `.tags` block:

```
.tags {
  .tag-datenstand-2020 {
    background-color: pink;
  }
  .tag-datenstand-2021 {
    background-color: blue;
  }
}
```

**Reply by christoph-LSN** [reply]
Thank you @brockfanning it works!

**Reply by brockfanning** [reply]
@christoph-LSN Sounds good. Also for your information, if you wanted the colors to be based on the order of the tags, instead of the content of the tags, CSS does offer an "nth-child" selector, so in theory you could have the first tag always yellow, the second tag always pink, the third tag always blue, etc.

**Reply by christoph-LSN** [reply]
@brockfanning that sounds intresting, but how do I do that nth-child?

adding this lines in my Custom.scss does have no effect

`.tags {
li:nth-child(1) {
  background: pink;
}
li:nth-child(2) {
  background: lightgreen;
}
li:nth-child(3) {
  background: yellow;
}
}`

**Reply by brockfanning** [reply]
@christoph-LSN That's close, but try something like this:

```
.tags {
  li:nth-child(1) .tag { background: pink; }
  li:nth-child(2) .tag { background: lightgreen; }
  li:nth-child(3) .tag { background: yellow; }
}
```

**Reply by christoph-LSN** [reply]
@brockfanning great it works,
but the first tag seems to be the "Daten verfügbar" tag and is still black. But this is no problem

![image](https://user-images.githubusercontent.com/94443931/174652227-1f42f814-d311-4a58-b1f0-4590bf0de763.png)
